tests/test_read.in

```diff
@@ -0,0 +1,55 @@
+#
+# Test the `read` builtin for non-interactive cases.
+#
+set -l read_limit (math '10 * 1024 * 1024')
+
+# Ensure the `read` command terminates if asked to read too much data. The var
+# should be empty. We throw away any data we read if it exceeds the limit on
+# what we consider reasonable.
+read x </dev/zero
```

Comment:
You'll want to output something before this line because this would hang without the limit.

`echo "Testing read limit. If this hangs, that's a bug!"`?